pkg/commands/git_config/get_key_test.go

```diff
@@ -0,0 +1,44 @@
+package git_config
+
+import (
+	"os"
+	"testing"
+)
+
+func TestGitConfigCount(t *testing.T) {
+	type scenario struct {
+		TestName             string
+		Envs                 map[string]string
+		ExpectedConfigValues map[string]string
+	}
+	scenarios := []scenario{
+		{
+			TestName: "1 config",
+			Envs: map[string]string{
+				"GIT_CONFIG_COUNT":   "1",
+				"GIT_CONFIG_KEY_0":   "user.signingkey",
+				"GIT_CONFIG_VALUE_0": "abc0123",
+			},
+			ExpectedConfigValues: map[string]string{
+				"user.signingkey": "abc0123",
+			},
+		},
+	}
+	for _, s := range scenarios {
+		t.Run(s.TestName, func(t *testing.T) {
+			for k, v := range s.Envs {
+				os.Setenv(k, v)
+			}
+			for k, v := range s.ExpectedConfigValues {
+				cmd := getGitConfigCmd(k)
+				res, err := runGitConfigCmd(cmd)
```

Comment:
I'm not really excited about running the test this way, because it uses your real git environment, which means that the test result could be influenced by whatever configs you have in your machine's global git config, our your lazygit repo's local config. I try to avoid this whenever possible, to make tests 100% deterministic. Not a huge deal in this case, because we know that the env vars will take precedence, but still.

That said, I'm not sure it's worth improving it in this case (and it's not immediately obvious how); just wanted to bring it up for discussion.